compat/src/index.js

```diff
@@ -122,6 +122,37 @@ const StrictMode = Fragment;
  */
 const flushSync = (callback, arg) => callback(arg);
 
+export function startTransition(cb) {
+	cb();
+}
+
+export function useDeferredValue(val) {
+	return val;
+}
+
+export function useTransition() {
+	return [false, startTransition];
+}
+
+// TODO: in theory this should be done after a VNode is diffed as we want to insert
+// styles/... before it attaches
+export const useInsertionEffect = useLayoutEffect;
+
+export function useSyncExternalStore(subscribe, getSnapshot) {
+	const [state, setState] = useState(getSnapshot);
+
+	// TODO: in suspense for data we could have a discrepancy here because Preact won't re-init the "useState"
+	// when this unsuspends which could lead to stale state as the subscription is torn down.
+
+	useEffect(() => {
+		return subscribe(() => {
+			setState(getSnapshot());
+		});
+	}, [subscribe, getSnapshot]);
```

Comment:
I came across this PR when I was looking into React 18 support in compat. Just wondering, shouldn't the state be set before subscribing? 🤔

```suggestion
	useEffect(() => {
	  const callback = () => setState(getSnapshot());
	  callback();
		return subscribe(callback);
	}, [subscribe, getSnapshot]);
```